crates/matrix-sdk/src/widget/filter.rs

```diff
@@ -103,271 +103,325 @@ pub enum StateEventFilter {
 }
 
 impl StateEventFilter {
-    fn matches(&self, matrix_event: &MatrixEventFilterInput) -> bool {
-        let Some(state_key) = &matrix_event.state_key else {
-            // Message-like event doesn't match a state event filter.
+    fn matches(&self, filter_input: &FilterInput) -> bool {
+        let FilterInput::State(state_filter_input) = filter_input else {
             return false;
         };
 
         match self {
-            StateEventFilter::WithType(event_type) => {
-                matrix_event.event_type == TimelineEventType::from(event_type.clone())
+            StateEventFilter::WithType(filter_type) => {
+                &state_filter_input.event_type == filter_type
             }
             StateEventFilter::WithTypeAndStateKey(event_type, filter_state_key) => {
-                matrix_event.event_type == TimelineEventType::from(event_type.clone())
-                    && state_key == filter_state_key
+                &state_filter_input.event_type == event_type
+                    && state_filter_input.state_key == *filter_state_key
             }
         }
     }
+    fn filter_event_type(&self) -> String {
```

Comment:
Why not return `&str`? I can see at least one case where you can avoid the copying.

Reply:
How can I construct `&str` from a `&StateEventType`. I think the only public method i have is `to_string()`

Reply:
Yeah, a bit annoying.

The overhead doesn't particularly matter I'd imagine, so you can pick whichever you prefer here.